packages/orchestration/src/internal.ts

```diff
@@ -0,0 +1,5 @@
+import type { Vow } from '@agoric/vow';
+
+export type PromiseToVow<T> = T extends (...args: infer A) => Promise<infer R>
```

Comment:
I'd like to see docs for `PromiseToVow` and type tests - at least 1 positive and 1 negative.

The name seems long. Maybe it doesn't need to appear in many places?

Reply:
Please see tests added here: [2dcc26f](https://github.com/Agoric/agoric-sdk/pull/9591/commits/2dcc26f29de7c2cbc1506f62293934341fa4dbaf). As always, thanks for careful review. Something was missed in the original implementation.

Reply:
"Missed" is maybe harsh. It would return `never` if the target was not a Promise. The caller shouldn't be using it in the first place. 

Now, it passes through all values and only transforms Promises to Vows.